src/cmdline/mod.rs

```diff
@@ -220,55 +402,41 @@ impl Cmdline {
                 .join(",")
         );
 
-        self.insert_str(kv_str)
+        self.insert_string(kv_str, args_type)
     }
 
-    /// Validates and inserts a string to the end of the current command line.
+    /// Validates and inserts a string as a new cmdline argument.
     ///
     /// # Arguments
     ///
-    /// * `slug` - String to be appended to the command line.
+    /// * `slug` - String to be appended to the args string.
+    /// * `args_type` - Type of the new argument to be inserted.
     ///
     /// # Examples
     ///
     /// ```rust
     /// # use linux_loader::cmdline::*;
-    /// # use std::ffi::CString;
-    /// let mut cl = Cmdline::new(100);
-    /// cl.insert_str("foobar");
-    /// let cl_cstring = CString::new(cl).unwrap();
-    /// assert_eq!(cl_cstring.to_str().unwrap(), "foobar");
+    /// let mut cl = Cmdline::new(10).unwrap();
+    /// cl.insert_string("foo=bar", CmdlineArgType::BootArg)
+    ///     .unwrap();
+    /// assert_eq!(cl.as_cstring().unwrap().as_bytes_with_nul(), b"foo=bar\0");
     /// ```
-    pub fn insert_str<T: AsRef<str>>(&mut self, slug: T) -> Result<()> {
-        let s = slug.as_ref();
+    pub fn insert_string<T: AsRef<str>>(
```

Comment:
Why change the name of this? It still receives something that can be a `str`, so the name was correct before.

Reply:
going back to the old naming